### PR TITLE
Add CheckedCPrepass

### PR DESCRIPTION
**Pull request by kkjeer**
This PR implements a prepass that traverses a function and gathers information that is used by Checked C analyses.

Currently, the pass gathers `VarUses`, which maps a `VarDecl` that is used in a bounds declaration or has declared bounds to the `DeclRefExpr` that is the first use of the `VarDecl`. This information is used by bounds checking, which passes it to `AbstractSetManager` in order to get or create an `AbstractSet` for a `VarDecl`.

In the future, this pass can be modified to gather additional information such as variables that are used in bounds, which can be used by the bounds widening analysis.

### CODE REVIEW

File: clang/include/clang/AST/AbstractSet.h [hunk 1]
```diff
@@ -75,16 +76,18 @@ namespace clang {
...
     Sema &S;
 
-    // VarUses maps a VarDecl with a bounds expression to the DeclRefExpr
-    // (if any) that is the first use of the VarDecl. If a VarDecl V has
-    // an entry in VarUses, the DeclRefExpr for V is used to get or create
-    // the AbstractSet for V. Otherwise, a use of V is constructed and
-    // added to VarUses. This created use of V is currently not released.
-    // It should be rare that a use of V needs to be created, since this
-    // should only occur if V is unused within the body of a function.
+    // VarUses maps a VarDecl to the DeclRefExpr (if any) that is the first
```
Comment (mgrang) [inline clang/include/clang/AST/AbstractSet.h:79]: that is the first first use of the VarDecl --> that is the first use of the VarDecl

**Review by mgrang — COMMENTED** [review]
I think the name `CheckedCPrepass` sounds too general as it indicates a compiler pre-pass as opposed to an analyses pre-pass. Maybe `CheckedCAnalysesPrePass` is a more apt name.

**Comment by sulekhark** [comment]
How about `CheckedCSemaPrepass`?

File: clang/lib/Sema/CheckedCAnalysesPrepass.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#include "clang/Sema/CheckedCAnalysesPrepass.h"
```
Comment (mgrang) [inline clang/lib/Sema/CheckedCAnalysesPrepass.cpp:1]: This file seems to be missing the header containing the license info.

**Review by mgrang — APPROVED** [review]
Thanks for addressing the review comments. LGTM!

**Pull request merged**